src/libraries/Common/src/Extensions/ProviderAliasUtilities/ProviderAliasUtilities.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.Extensions.Logging
 {
     internal static class ProviderAliasUtilities
     {
-        private const string AliasAttributeTypeFullName = "Microsoft.Extensions.Logging.ProviderAliasAttribute";
+        private static readonly string AliasAttributeTypeFullName = typeof(ProviderAliasAttribute).FullName!;
```

Comment:
This is just an expensive way to compute the constant string. Is it really worth it?

It is very common to hardcode type names as strings. You can find many instances of hardcoded type name strings in this repo.

Reply:
It's a single call, done only once. Do you think this makes much of a difference?

Reply:
I would say yes. This might root the reflection stack on Native AOT.

Reply:
> Do you think this makes much of a difference?

One-off cases won't make a difference. If we were to use pattern like this everywhere instead of hardcoded strings, it would make a difference.

Reply:
In other words, this is inconsistent with what we typically do in the rest of the repo.

Reply:
I gave it a second thought: Since `ProviderAliasAttribute` is statically linked in this project, there won't be any other class definition with this FQ name.

So, it may be logical and coherent to replace the const string comparison

```c#
if (attributeData.AttributeType.FullName == AliasAttributeTypeFullName &&
```

in [line 22](https://github.com/dotnet/runtime/blob/159f40e850839050ac09cb419d9830adeab22e2b/src/libraries/Common/src/Extensions/ProviderAliasUtilities/ProviderAliasUtilities.cs#L22) with an `is` operator

```c#
if (attributeData.AttributeType is ProviderAliasAttribute
```

What do you think?

Reply:
I forgot to mention that I very much agree with your valuable AOT argument.

I'm not sure if the `is` operator is made available for all .NET versions.

This option I provided just as an option. We can easily refrain from this change and move over to PR #114606. It contains all the changes I made without the string comparison being changed.

Reply:
I just reopened #114606. Please choose.